### PR TITLE
Spruce up the emoji renderer

### PR DESCRIPTION
**Pull request by davidwengier**
-trees parameter to control number of trrees output
-trains parameter to control number of trains output
Trains have consistent colours across generation
Generate trains on bridges, and on nothing
Generate multiple sizes of emoji, all with consistent colours and sizes

### CODE REVIEW

File: src/Trains.Emoji/Program.cs [hunk 1]
```diff
@@ -10,80 +11,141 @@ namespace Trains.Emoji
...
+        private const string BaseFolderName = "EmojiOutput";
 
-        public static void Main()
-            => DI.ServiceLocator.GetService<EmojiDrawer>().Save(512);
+        public static void Main(string[] args)
+        {
+            var imageSizes = new List<int>();
 
-        public EmojiDrawer(IRenderer<Tree> tree, IRenderer<Track> track, IRenderer<Train> train, IPixelMapper pixelMapper)
+            for (int i = 0; i < args.Length; i++)
```
Comment (ScottHolden) [inline src/Trains.Emoji/Program.cs:28]: Wasn't there a simple MS library for this? More of an open question!

File: src/Trains.Emoji/Program.cs [hunk 2]
```diff
@@ -10,80 +11,141 @@ namespace Trains.Emoji
...
+
+                if (!Directory.Exists(folderName))
+                {
+                    Directory.CreateDirectory(folderName);
+                }
+                for (int i = 0; i < s_numberOfTreesToDraw; i++)
+                {
+                    Draw("tree" + i, folderName, pixelMapper, x => _tree.Render(x, new Tree() { Seed = i }));
+                }
+                foreach (TrackDirection direction in (TrackDirection[])Enum.GetValues(typeof(TrackDirection)))
```
Comment (ScottHolden) [inline src/Trains.Emoji/Program.cs:84]: Wrap lines 84-95 in a loop and do it with both just the track, and bridge+track?

File: src/Trains.Emoji/Program.cs [hunk 3]
```diff
@@ -10,80 +11,141 @@ namespace Trains.Emoji
...
-                float scale = _pixelMapper.CellSize / 100.0f;
+                float scale = pixelMapper.CellSize / 100.0f;
                 canvas.Scale(scale, scale);
 
                 x(canvas);
 
                 canvas.Restore();
             }
-            using Stream s = File.OpenWrite(FolderName + "\\" + name + ".png");
+            using Stream s = File.OpenWrite(Path.Combine(folderName, name + ".png"));
```
Comment (ScottHolden) [inline src/Trains.Emoji/Program.cs:136]: <3

File: src/Trains.Emoji/Program.cs (hunk 1 above)
Comment (davidwengier) [reply src/Trains.Emoji/Program.cs:28]: Yes, there is. I think upgrading to it makes for good stream content :)

File: src/Trains.Emoji/Program.cs (hunk 2 above)
Comment (davidwengier) [reply src/Trains.Emoji/Program.cs:84]: Lots more clean up to do for sure, but it was late 😛

**Pull request merged**